### PR TITLE
Add BFCM promo

### PR DESCRIPTION
**Pull request by bor0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Implement BFCM in-app promo.

⚠️ After merging, make sure to update `wccom_iam_promos` option on WCCOM with the necessary data (see [this](21881-gh-Automattic/woocommerce.com)). ⚠️

Closes 21814-gh-Automattic/woocommerce.com.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

#### Backward compatibility I

1. Switch to `trunk`
2. Ensure the [Marketplace endpoint](https://github.com/woocommerce/woocommerce/blob/f9cf25e/plugins/woocommerce/includes/admin/class-wc-admin-assets.php#L562) returns BFCM data (i.e. apply [this patch](https://github.com/user-attachments/files/17175811/test.patch))
3. Open the following screens, make sure the frontend code still works:
    - `WooCommerce > Extensions` (make sure you see the weekly deals notice (if you don't, run `localStorage.removeItem( 'wc-marketplaceNoticeClosed-promotion-0' )`))
    - `WooCommerce > Home` (make sure screen renders)
    - `Marketing > Overview` (make sure screen renders)
    - `Marketing > Coupons` (make sure screen renders)

#### Backward compatibility II

1. Switch to this branch
2. Ensure the [Marketplace endpoint](https://github.com/woocommerce/woocommerce/blob/f9cf25e/plugins/woocommerce/includes/admin/class-wc-admin-assets.php#L562) doesn't return BFCM data (i.e. the data in [this patch](https://github.com/user-attachments/files/17175811/test.patch) is **not** returned)
3. Open the following screens, make sure the frontend code still works:
    - `WooCommerce > Extensions` (make sure you see the weekly deals notice (if you don't, run `localStorage.removeItem( 'wc-marketplaceNoticeClosed-promotion-0' )`))
    - `WooCommerce > Home` (make sure screen renders)
    - `Marketing > Overview` (make sure screen renders)
    - `Marketing > Coupons` (make sure screen renders)

#### New functionality

1. Apply [test.patch](https://github.com/user-attachments/files/17175811/test.patch) locally
2. Do `localStorage.setItem( 'debug', 'wc-admin:*' );` in your JS console to show Track events. Refresh the page, and events should show up in your JS console like this. (If you're using Chrome, enable verbose messages in the console.) 

image

3. For each of `WooCommerce > Extensions`, `WooCommerce > Home`, `Marketing > Overview`, `Marketing > Coupons`:
    - Ensure you see the promo card for Black Friday. Track event `wcadmin_marketplace_promotion_viewed` should fire – check the props look OK.
    - Ensure link works. Track event `wcadmin_marketplace_promotion_actioned` should fire.
    - Ensure dismiss works (and doesn't show after reload). Track event `wcadmin_marketplace_promotion_dismissed` should fire.
    - To reset dismiss state for all of these four screens, run `localStorage.removeItem( 'wc-marketplaceDismissedPromos' )`

#### Bonus points: integration with WCCOM

1. Revert the patch you applied before.
2. Install [WP Crontrol](https://wordpress.org/plugins/wp-crontrol/) on your WooCommerce environment.
5. Set up your local WCCOM as in the test steps for 21881-gh-Automattic/woocommerce.com.
6. Apply this patch: [test-api.patch](https://github.com/user-attachments/files/17226716/test-api.patch). This points your WooCommerce environment at your local WCCOM.
7. Find and run the `woocommerce_marketplace_cron_fetch_promotions` cron event in `Tools > Cron Events` (example URL `http://localhost:8888/wp-admin/tools.php?page=crontrol_admin_manage_page&s=woocommerce_marketplace_cron_fetch_promotions`). This should update the `woocommerce_marketplace_promotions_v2` transient with the data from WCCOM.
8. Check that the promos from `New functionality` appear.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/marketplace/components/promotions/promotions.tsx [hunk 1]
```diff
@@ -12,37 +14,24 @@ declare global {
...
-
-type Page = {
-	page: string;
-	path: string;
-	tab?: string;
-};
-
-const Promotions: () => null | JSX.Element = () => {
+const Promotions: ( {
+	promoCardOnly,
```
Comment (bor0) [inline plugins/woocommerce-admin/client/marketplace/components/promotions/promotions.tsx]: This feels like a dirty hack, but it's mainly there to preserve BWC. Ideally, each ad would use the API data format (and dismiss logic) as introduced here.

File: plugins/woocommerce-admin/client/marketplace/components/promotions/promotions.tsx [hunk 2]
```diff
@@ -12,37 +14,21 @@ declare global {
...
-	path: string;
-	tab?: string;
-};
-
-const Promotions: () => null | JSX.Element = () => {
+const Promotions: ( {
+	format,
+}: {
+	format?: string;
+} ) => null | JSX.Element = ( { format = 'notice' } ) => {
```
Comment (bor0) [inline plugins/woocommerce-admin/client/marketplace/components/promotions/promotions.tsx]: @andfinally, thank you for this commit! It's 💯.

My only proposal is to make the argument required.

```suggestion
	format: string;
} ) => null | JSX.Element = ( { format } ) => {
```

File: plugins/woocommerce-admin/client/marketplace/components/promotions/promotions.tsx (hunk 1 above)
Comment (bor0) [reply plugins/woocommerce-admin/client/marketplace/components/promotions/promotions.tsx]: Addressed in 80078f8, much neater. Props @andfinally

File: plugins/woocommerce-admin/client/marketplace/components/promotions/promotions.tsx (hunk 2 above)
Comment (andfinally) [reply plugins/woocommerce-admin/client/marketplace/components/promotions/promotions.tsx]: Thanks Boro, committed.

**Review by raja651 — COMMENTED** [review]
@bor0  and @andfinally I can see the BCFM promo card on Home, extensions, marketing, coupons correctly. But i don't see the tracks event registering . May be i'm looking at the wrong place i checked the network tab on browser after applying this setting `localStorage.setItem( 'debug', 'wc-admin:*' );` according to test instructions.

Screenshot 2024-10-03 at 2 51 23 PM
Screenshot 2024-10-03 at 2 50 45 PM

**Comment by andfinally** [comment]
Thanks a lot @raja651, I'll have a look. I've just got a bunch of design feedback from Poli, will make some tweaks in the styling as well.

**Comment by andfinally** [comment]
Hi @raja651, just wondering, did you look in your JS console for the Tracks events? I've updated the test steps to clarify.

**Review by raja651 — APPROVED** [review]
Tracks are shown now, it was hidden on my console !

**Review by andfinally — APPROVED** [review]
This looks and tests as expected.

**Pull request merged**